src/disposable.coffee

```diff
@@ -24,7 +24,9 @@ class Disposable
   dispose: ->
     unless @disposed
       @disposed = true
-      @disposalAction?()
```

Comment:
>  I don't think it's good for Disposable to accept something that is not a function

I agree, but it shouldn't blow up here (during disposal which may never happen), it should blow up when the disposable is created.